en_US/modules/lwm2m_protocol.md

```diff
@@ -2,80 +2,89 @@
+so the message structure is simple and compact to worker better in environments where network resources are limited and the device is always online.
 
-LwM2M Server serves as a server and is deployed at the M2M service provider or network service provider. LwM2M defines two types of servers
+There are two roles in the LwM2M protocol:
 
-  -One is LwM2M BOOTSTRAP SERVER. The emqx-lwm2m plug-in does not implement the server function.
-  -One is LwM2M SERVER, emqx-lwm2m realizes the function of this server on UDP, but SMS does not.
+- LwM2M Server
 
-LwM2M Client acts as a client and is deployed on each LwM2M device.
+  LwM2M has two kind services and deployed at the M2M service providers or network service providers:
+
+  1. LwM2M bootstrap server, emqx-lwm2m plugin does not implement this server.
+  2. LwM2M server, emqx-lwm2m plugin implements the function of this server on UDP, while SMS does not.
+
+- LwM2M Client
+
+    LwM2M Client acts as a client and is deployed on each LwM2M device.
 
 Between LwM2M Server and LwM2M Client, the LwM2M protocol defines 4 interfaces.
 
-1. Bootstrap interface Bootstrap: provide LwM2M client with registration to LwM2M
-    The necessary information of the server, such as server access information, resource information supported by the client, etc.
-2. Client Registration Interface Client Registration: Connect LwM2M client and LwM2M server, connect LwM2M
-    The relevant information of the client is stored on the LwM2M server. Only after completing the registration, LwM2M
-    The communication and management between the client and the server becomes possible.
-3. Device management and service implementation interface Device Management and Service Enablement: the master of this interface is LwM2M
-    The server sends instructions to the client, and the client responds to the instructions and sends the response message to the server.
-4. Information reporting interface Information Reporting: Allow LwM2M
-    The server subscribes to the client for resource information, and the client receives the subscription and reports its resource changes to the server according to the agreed mode.
+1. Bootstrap interface
+
+    Provide the LwM2M client with the necessary information to register with the LwM2M server, such as sever access information, resource information supported by the client, etc.
+
+2. Client Registration Interface
+
+    Connect the LwM2M client and the LwM2M server, and stored the information about connection of the LwM2M client. Communication and management between LwM2M client and server is only possible after registration is completed.
+
+3. Device management and service implementation interface
+
+    Allows the server to access and modify LwM2M client object instances and resources.
+
+4. Information reporting interface
+
+    Allows the server to subscribe resource information to the client, and the client reports its resource changes to the server in the agreed mode.
```

Comment:
there was no "agreed mode" in previous context, so the 'the' lacks of a context, suggestion is to remove it.
```suggestion
    Allows the server to subscribe resource information to the client, and the client reports its resource changes to the server in agreed mode.
```